src/nvidia.cc

```diff
@@ -827,27 +888,37 @@ void print_nvidia_value(struct text_object *obj, char *p, int p_max_size)
 	struct nvidia_s *nvs = static_cast<nvidia_s *>(obj->data.opaque);
 	int value, temp1, temp2;
 	char* str;
+	
+    Display *dpy = nvdisplay ? nvdisplay : display;
+    
+    //num_GPU calculated only once based on the physical target
+	static int num_GPU = get_nvidia_target_count(dpy, TARGET_GPU) - 1;
```

Comment:
When I did it I  I was seeing my cards as an array zero based. Then it was more logical for me to have it that way.

But I agree that can make more sense when you see num_GPU you expect to have at least one. Therefore I ahve to move the -1 operation when reading the number of GPU as you have seen.

I change it.